src/geom_handlers/site_assessment.jl

```diff
@@ -60,5 +61,327 @@ function assess_reef_site(rst, geom, ruleset; degree_step=15.0)
+            continue
+        end
+
+        window = trues(size(c_rst))
+        for (n, crit_rule) in ruleset
+            window .= window .& crit_rule(c_rst[n])
+            if count(window) < ceil(length(window) / 3)
+                # Stop checking other rules if below hard threshold
+                break
+            end
+        end
+
+        score[j] = mean(window)
+        best_poly[j] = rot_geom
+    end
+
+    return score[argmax(score)], argmax(score), best_poly[argmax(score)]
+end
+
+"""
+    identify_potential_sites_edges(
+        rst_stack::RasterStack,
+        indices_pixels::Raster,
+        indices::Vector{CartesianIndex{2}},
+        gdf::DataFrame,
+        x_dist::Union{Int64, Float64},
+        y_dist::Union{Int64, Float64},
+        target_crs::GeoFormatTypes.CoordinateReferenceSystemFormat,
+        reef_lines::Vector{Vector{GeometryBasics.Line{2, Float64}}};
+        degree_step::Float64=15.0,
+        n_rot_per_side::Int64=2
+    )::DataFrame
+
+Identify the most suitable site polygons for each pixel in the `indices_pixels` raster where
+`indices` denotes which pixels to check for suitability. `x_dist` and `y_dist` are x and y
+lengths of the search polygon. A buffer of `rst_stack` resolution is applied to the search box.
+And angle from a pixel to a reef edge is identified and used for searching with custom rotation
+parameters.
+
+# Arguments
+- `rst_stack` : RasterStack containing environmental variables for assessment.
+- `indices_pixels` : Raster that matches indices for lon/lat information.
+- `indices` : Vector of CartesianIndices noting pixels to assess sites.
+- `gdf` : GeoDataFrame containing the reef outlines used to align the search box edge.
+- `x_dist` : Length of horizontal side of search box.
+- `y_dist` : Length of vertical side of search box.
+- `target_crs` : CRS of the input Rasters. Using GeoFormatTypes.EPSG().
+- `reef_lines` : Vector containing reef outline segments for each reef in `gdf`.
+- `degree_step` : Degree to perform rotations around identified edge angle.
+- `n_rot_per_side` : Number of rotations to perform clockwise and anticlockwise around the identified edge angle. Default 2 rotations.
+
+# Returns
+DataFrame containing highest score, rotation and polygon for each assessment at pixels in indices.
+"""
+function identify_potential_sites_edges(
+    rst_stack::RasterStack,
+    indices_pixels::Raster,
+    indices::Vector{CartesianIndex{2}},
+    gdf::DataFrame,
+    x_dist::Union{Int64,Float64},
+    y_dist::Union{Int64,Float64},
+    target_crs::GeoFormatTypes.CoordinateReferenceSystemFormat,
+    reef_lines::Vector{Vector{GeometryBasics.Line{2,Float64}}};
+    degree_step::Float64=15.0,
+    n_rot_per_side::Int64=2
+)::DataFrame
+    res = abs(step(dims(rst_stack, X)))
+
+    # # TODO: Dynamically build this ruleset
+    ruleset = Dict(
+        :Depth => (data) -> within_thresholds(data, -9.0, -2.0),
+        :WavesTp => (data) -> within_thresholds(data, 0.0, 5.9)
+    )
```

Comment:
Was this left in intentionally?

Reply:
Yeah I just wasn't sure if that was a work in progress you were working on for user defined thresholds or if I just had to add all the criteria in.

Reply:
It'll be an argument passed in. We can leave as is for now and put all the pieces together soon